include/envoy/http/conn_pool.h

```diff
@@ -116,6 +116,11 @@ class Instance : public Event::DeferredDeletable {
    *                      should be done by resetting the stream.
    */
   virtual Cancellable* newStream(Http::StreamDecoder& response_decoder, Callbacks& callbacks) PURE;
+
+  /**
+   * @return Upstream::HostConstSharedPtr the host for which connections are pooled.
+   */
+  virtual Upstream::HostConstSharedPtr host() const PURE;
```

Comment:
Added so we can return hostname and address as an additional debug response header.

Reply:
Do we need this? See `onUpstreamHostSelected` where we store the host inside stream info?

Edit: I guess you need this before the callbacks come in. I'm wondering if you could get rid of the callbacks potentially with this method and set it directly? Worth thinking about.

Reply:
Yeah, I don't think I have any other way of extracting the host info from the connection pool until `onPoolReady` or `onPoolFailure` fire.

I'm trying to figure out if it's safe for me to move the do_not_forward logic out from `Filter::decodeHeaders`, down into `Filter::UpstreamRequest::onPoolReady`. If I can do that, then I can get rid of the new `host` accessor on the connection pool...

Reply:
Update: I couldn't work out how to get rid of this. It seems like by the time we hit `onPoolReady`, where the host is handed to us, we've already done a bunch of work we'd have wanted to avoid. Let me know if this is ok, @mattklein123, or if I'm missing a better way.

Reply:
If you need to keep it that's fine, I can take another look in final review but it's not a big deal.